### PR TITLE
Add path to Keystore

### PR DESCRIPTION
**Pull request by rootulp**
**What type of PR is this?**

> Bug fix

**What does this PR do? Why is it needed?**

Adds `path` to the `Keystore` struct because it is a required JSON field per [EIP-2335](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2335.md#json-schema)

**Which issues(s) does this PR fix?** 

Fixes https://github.com/prysmaticlabs/prysm/issues/10107

### CODE REVIEW

File: validator/keymanager/types_test.go [hunk 1]
```diff
@@ -20,3 +24,12 @@ var (
...
 	_ = keymanager.Deleter(&derived.Keymanager{})
 )
+
+func TestKeystoreContainsPath(t *testing.T) {
+	keystore := keymanager.Keystore{}
+	encoded, err := json.Marshal(keystore)
+	want := "{\"crypto\":null,\"uuid\":\"\",\"pubkey\":\"\",\"version\":0,\"name\":\"\",\"path\":\"\"}"
+
+	require.NoError(t, err, "Unexpected error marshalling keystore")
+	require.Equal(t, want, string(encoded))
```
Comment (rootulp) [inline validator/keymanager/types_test.go:34]: I considered replacing this assertion with

```go
if !strings.Contains(string(encoded), "path") {
    t.Errorf("Encoded %s does not contain path", encoded)
}
```

which would remove line 31. Defer to reviewers on what they think is clearer

Comment (rkapka) [reply validator/keymanager/types_test.go:34]: `assert.Equal(t, true, strings.Contains(string(encoded), "path"))`

**Comment by rootulp** [comment]
Thanks for the feedback @rkapka. I addressed so this PR is ready for re-review.

**Review by james-prysm — COMMENTED** [review]
looks like you need to run the gazelle prefix "github.com/prysmaticlabs/prysm/validator/keymanager" command in the validators keymanager folder to fix the linter, must be from added dependencies for test. thanks for the contribution LGTM after that.

**Comment by rootulp** [comment]
Thanks for the tip @james-prysm !

On a slightly related note, `brew install bazel` installs bazel 5.0.0 and it looks like this project is using `4.2.2` per
https://github.com/prysmaticlabs/prysm/blob/4b3a7231661e4eb97cbfa70694859607efb8e82c/.bazelversion#L1

Should I open a new issue to track upgrading to bazel `5.0.0` because it doesn't look like such an issue exists yet ([search](https://github.com/prysmaticlabs/prysm/issues?q=is%3Aissue+is%3Aopen+bazel+5.0.0)).

**Comment by james-prysm** [comment]
> Thanks for the tip @james-prysm !
> 
> On a slightly related note, `brew install bazel` installs bazel 5.0.0 and it looks like this project is using `4.2.2` per
> 
> https://github.com/prysmaticlabs/prysm/blob/4b3a7231661e4eb97cbfa70694859607efb8e82c/.bazelversion#L1
> 
> Should I open a new issue to track upgrading to bazel `5.0.0` because it doesn't look like such an issue exists yet ([search](https://github.com/prysmaticlabs/prysm/issues?q=is%3Aissue+is%3Aopen+bazel+5.0.0)).

let me double check with the team I'm sure they know about bazel 5.0.0 but might not be ready to upgrade yet ( assumption) i'd hold off on it. happy to continue the conversation on discord if interested.

**Comment by rootulp** [comment]
Sounds good, I'll hold off on creating a GH issue.

**Comment by rkapka** [comment]
This PR should definitely *not* update Bazel.

**Comment by rootulp** [comment]
> This PR should definitely _not_ update Bazel.

Agreed. My intention was to create a separate GH issue to track upgrading to Bazel 5.0.0 but defer to you / the rest of the maintainers on how best to track that effort.

Thanks for reviewing and merging! Keep up the great work on Prysm.

**Pull request merged**